### PR TITLE
Redesign clauses

### PR DESCRIPTION
**Pull request by dairyo**
According to KiiPlatform/thing-if-AndroidSDK#179, I redesigned clauses.

I will remove old clauses and change APIs using old clauses in other PR.

Related PR: #265

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/QueryClause.swift [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+     - Parameter upper: upper limit
+       - limit: Limit value. The type must be integer, double or
+         float.
+       - included: Includes limit value or not. If true, limit value
+         is included. If false, limit value is not included
+     - Returns: An instance of `RangeClauseInQuery`.
+     */
+    open static func range(
+      _ field: String,
+      lower: (limit: AnyObject, included: Bool),
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/QueryClause.swift]: lowerLimit should not be `AnyObject`, better to limit it to number types(Int, Double, Float)

File: ThingIFSDK/ThingIFSDK/QueryClause.swift [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+      upper: (limit: AnyObject, included: Bool)) -> RangeClauseInQuery
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/QueryClause.swift]: upperLimit should not be `AnyObject`

**Comment by dairyo** [comment]
@yongpingchen I changed type of limit from AnyObject to NSNumber at 1449467.

File: ThingIFSDK/ThingIFSDK/QueryClause.swift [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+//
+
+import Foundation
+
+/** Base class for query clause classes.
+
+ This is abstract class. Developers does not need to directly use this
+ class.
+ */
+open class QueryClause: NSObject, BaseClause {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/QueryClause.swift:16]: I think `QueryClause` should be `protocol`.

File: ThingIFSDK/ThingIFSDK/TriggerClause.swift [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+//
+
+import Foundation
+
+/** Base class for trigger clause classes.
+
+ This is abstract class. Developers does not need to directly use this
+ class.
+ */
+open class TriggerClause: NSObject, BaseClause {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/TriggerClause.swift:16]: I think `TriggerClause` should be `protocol`

File: ThingIFSDK/ThingIFSDK/QueryClause.swift (hunk 3 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/QueryClause.swift:16]: I want to do so but swift does not allow it :(

Swift does not allow to use `Protocol` as type parameter.
`AndClauseInQuery` and `OrClauseInQuery` use `QueryClause` as type parameter by `typealias`. We need concrete class.

In my knowledge, this is better way.

What do you think?

File: ThingIFSDK/ThingIFSDK/TriggerClause.swift (hunk 4 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/TriggerClause.swift:16]: This case is same as `QueryTrigger`. We need concrete class.

File: ThingIFSDK/ThingIFSDK/QueryClause.swift (hunk 3 above)
Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/QueryClause.swift:16]: I see. It is okay then. You already made the call of methods of these classes failure.

**Comment by yongpingchen** [comment]
LGTM

**Pull request merged**